units/provisioning_agent_deploy.go

```diff
@@ -205,9 +205,13 @@ func (j *agentDeployJob) startAgentOnHost(ctx context.Context, settings *evergre
 		})
 		return nil
 	}
-
 	grip.Info(message.Fields{"runner": "taskrunner", "message": "prepping host finished successfully", "host": hostObj.Id})
 
+	// stop here if the host is a parent for containers - don't want the agent running
+	if hostObj.HasContainers {
+		return nil
```

Comment:
Is there anything else that needs to happen here, e.g., `SetNeedsNewAgent(false)` and other things that happen later in the job?

Reply:
good point, changed so that the only thing we don't do is start the agent and set a host secret